### DISCUSSION TITLE
How to keep the vertical scroll bar of the MultiLineEntry component always at the bot

### DISCUSSION THREAD

**Discussion by EddieChan1993** [fyne-io/fyne#3797] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

How to keep the vertical scroll bar of the MultiLineEntry component always at the bot

### How to reproduce

How to keep the vertical scroll bar of the MultiLineEntry component always at the bot

### Example code

How to keep the vertical scroll bar of the MultiLineEntry component always at the bot

### Fyne version

v2.3.1

### Go compiler version

1.18

### Operating system and version

macOs

**Comment by andydotxyz** [comment]
I'm not sure what this question means. The vertical scroll bar displays where the user is through scrolled content of the multiline entry - so it needs to be the height of the input.

Perhaps an illustration of what you are seeing vs what is desired would help?

**Reply by EddieChan1993** [reply]
When the text content exceeds the boundaries of this component, a vertical scrollbar appears. However, I want this scrollbar to default to the bottom position so that I can see the bottommost text content.

**Reply by andydotxyz** [reply]
The scrollbar will follow the cursor for user input.
If you are manually setting the text consider updating the cursor as well.

**Reply by EddieChan1993** [reply]
how update the cursor

**Reply by Jacalz** [reply]
Check the API for entry over at https://github.com/fyne-io/fyne/blob/v2.3.3/widget/entry.go#L37. You have the `CursorRow` and `CursorColumn` fields that you can set and then call `.Refresh()`.

**Reply by EddieChan1993** [reply]
thanks

**Reply by EddieChan1993** [reply]
I called the DragEnd method interface but it didn't meet my expectations. Could you please advise me on how to write this section of the code?

**Reply by Jacalz** [reply]
FYI: I meant to link to https://pkg.go.dev/fyne.io/fyne/v2/widget#Entry and not the code on GitHub above.

You should not call `DragEnd()`. It is called internally by the driver to tell the widget that a drag has ended, not to drag something to the end. See the explanation in the my comment above about how to change the cursor position.

**Reply by EddieChan1993** [reply]
3ks

**Reply by EddieChan1993** [reply]
Currently, the SetText method of the MultiLineEntry component sets the scroll bar to the top by default when the content passed in is too much. I need to keep the scroll bar at the bottom so that I can see the latest content,this is the question I want to ask.

**Reply by andydotxyz** [reply, marked as answer]
> Check the API for entry over at https://github.com/fyne-io/fyne/blob/v2.3.3/widget/entry.go#L37. You have the `CursorRow` and `CursorColumn` fields that you can set and then call `.Refresh()`.

This is the answer you need

**Reply by EddieChan1993** [reply]
这个的前提是，光标必须在该组件上，不然没用

**Reply by andydotxyz** [reply]
Google translate:

The premise of this is that the cursor must be on the component, otherwise it is useless

**Reply by andydotxyz** [reply]
I cannot understand. The cursor refers to the line/bar/indicator inside the entry. You can move it as @Jacalz said.

**Reply by EddieChan1993** [reply]
Thank you very much, I have resolved this issue. Here is the link to my project https://github.com/EddieChan1993/chatgptgo
